src/Strategy/ManifestStrategy.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace Humbug\SelfUpdate\Strategy;
+
+use Humbug\SelfUpdate\Exception\HttpRequestException;
+use Humbug\SelfUpdate\Exception\JsonParsingException;
+use Humbug\SelfUpdate\Updater;
+use Humbug\SelfUpdate\VersionParser;
+
+class ManifestStrategy implements StrategyInterface
+{
+    /** @var array */
+    private static $requiredKeys = array('sha1', 'version', 'url');
+    /** @var string */
+    private $manifestUrl;
+    /** @var array */
+    private $manifest;
+    /** @var array */
+    private $availableVersions;
+    /** @var string */
+    private $localVersion;
+    /** @var bool */
+    private $allowMajor = false;
+    /** @var bool */
+    private $allowUnstable = false;
+
+    /**
+     * ManifestStrategy constructor.
```

Comment:
Opinions can vary, but I'm fairly verbose myself in comments. I just find it a PITA, esp. in open source projects, to spend too time familiarising myself with how something works so a bit of hand holding can be welcome.

Also, easy to copy/paste to a README and THEN delete the comments ;).